src/command_modules/azure-cli-webapp/azure/cli/command_modules/webapp/custom.py

```diff
@@ -0,0 +1,332 @@
+                                     http_logs=http_logs,
+                                     failed_requests_tracing=failed_request_tracing_logs,
+                                     detailed_error_messages=detailed_error_messages_logs)
+    if slot is None:
+        return client.sites.update_site_logs_config(resource_group, name, site_log_config)
+    else:
+        return client.sites.update_site_logs_config_slot(resource_group, name,
+                                                         site_log_config, slot)
+
+def config_slot_auto_swap(resource_group, webapp, slot, auto_swap_slot=None, disable=None):
+    client = web_client_factory()
+    site_config = client.sites.get_site_config_slot(resource_group, webapp, slot)
+    site_config.auto_swap_slot_name = '' if disable else (auto_swap_slot or 'production')
+    return client.sites.update_site_config_slot(resource_group, webapp, site_config, slot)
+
+def get_streaming_log(resource_group, name, provider=None, slot=None):
+    client = web_client_factory()
+    repo_url = _get_repo_url(client, resource_group, name, slot)
+    streaming_url = repo_url + '/logstream'
+    import time
+    if provider:
+        streaming_url += ('/' + provider.lstrip('/'))
+
+    user, password = _get_site_credential(client, resource_group, name)
+    t = CurlThread(streaming_url, user, password)
+    t.daemon = True
+    t.start()
+
+    while True:
+        time.sleep(100) #so that ctrl+c can stop the command
+
+def download_historical_logs(resource_group, name, log_file=None, slot=None):
+    '''
+    Download historical logs as a zip file
+    '''
+    client = web_client_factory()
+    user, password = _get_site_credential(client, resource_group, name)
+    repo_url = _get_repo_url(client, resource_group, name, slot)
+    host = urlparse(repo_url).netloc
+    url = "https://{}:{}@{}/dump".format(user, password, host)
+    import requests
+    r = requests.get(url, stream=True)
+    with open(log_file, 'wb') as f:
+        for chunk in r.iter_content(chunk_size=1024):
+            if chunk: # filter out keep-alive new chunks
+                f.write(chunk)
+    logger.warning('Downloaded logs to %s', log_file)
+
+def _get_site_credential(client, resource_group, name):
+    creds = client.sites.list_site_publishing_credentials(resource_group, name)
+    creds = creds.result()
+    return (creds.publishing_user_name, creds.publishing_password)
+
+class CurlThread(threading.Thread):
+    def __init__(self, stream_url, user_name, password):
+        threading.Thread.__init__(self)
+        self.streaming_url = stream_url
+        self.user_name = user_name
+        self.password = password
+
+    def run(self):
+        import pycurl
+        import certifi
+        c = pycurl.Curl()
+        c.setopt(c.URL, self.streaming_url)
+        c.setopt(pycurl.CAINFO, certifi.where())
+        c.setopt(c.USERPWD, '{}:{}'.format(self.user_name, self.password))
+        c.perform()
```

Comment:
I will change to a function like you recommended(have not found established pattern on this). There is a limitation that i pass arguments through a tuple, which is fine since i will not have more than 3 :-)